activeness:
  frame_id:
    fixed: $(arg robot_namespace)/map_locus

  planning_tree_resolution: 0.4

  submap_distance: 20.0 # [m] from the UAV into all directions

  main_timer:
    rate: 20.0 # [Hz]

  diagnostics_timer:
    rate: 10.0 # [Hz]

  euclidean_distance_cutoff: 1.4 # [m] maximum distance for Euclidean distance field
  permissible_distance_cutoff: 2.0 # [m] minimum distance to consider success
  
  safe_obstacle_distance: 1.2 # [m]

  min_path_length: 0.2 # [m] minimum length of path to be published

  distance_penalty: 1.0 # coefficient for traveled distance from start in A* cost function
  greedy_penalty: 2.0 # coefficient for estimated distance to goal in A* cost function

  max_waypoint_distance: 0.0 # [m] // or set negative for no padding
  min_altitude: -1000.0 # [m]
  max_altitude: 3.0 # [m]
  unknown_is_occupied: false # if false, planning through unknown space is enabled  #FOR SOLAR FARM
  # unknown_is_occupied: true # if false, planning through unknown space is enabled 
  replan_after: 1.0 # [s] time from last planning after which replanning is triggered

  timeout_threshold: 2.0 # [s] time after which A* planning is aborted and returns path to the node with the so far lowest cost   
  time_for_trajectory_generator: 0.6 # [s]

  # future collision check
  future_check_timer: # timer for collision checking with most recent map 
    rate: 10.0 # [Hz]

  collision_check_point_count: 8 # [-] number of points used for future collision check of UAV trajectory, uniformly sampled on a 2D circle around each trajectory point
  min_allowed_trajectory_points_after_crop: 10 # [-] when future trajectory approaches an obstacle, it can be cropped up to this number of points

  trajectory_generator:
    input_trajectory_length: 12.0 # [s] (the rest of path will be ignored during trajectory generation)
    relax_heading: true # if true, constraints on heading dynamics are ignored during trajectory generation process
    use_heading: false # if false heading is ignored during trajectory tracking
    heading_offset: 0.0 # [rad] offset of desired heading to direction of flight
    turn_in_flight_direction: false # sample heading reference in direction of flight
    max_segment_length_heading: 1.0 # [m] maximum length of segment for heading sampling